src/point-vector-fallbacks.jl

```diff
@@ -0,0 +1,194 @@
+
+"""
+    manifold_thing_forwards(T::Symbol, field::Symbol)
+
+Introduce basic fallbacks for type `T` that represents points or vectors for a manifold.
+Fallbacks will work by forwarding to field passed in the second argument.
+
+List of forwarded functions:
+* [`allocate`](@ref),
+* [`copy`](@ref),
+* [`copyto!`](@ref),
+* [`number_eltype`](@ref) (only for values, not the type itself),
+* `similar`,
+* `==`.
+"""
+macro manifold_thing_forwards(T, Twhere, field::Symbol)
```

Comment:
I will probably drop the `_thing` here. `manifold_fallbacks` sounds fair enough, “thing” is too unspecific, if we need a third word I would opt for `_element`. And are the docs right that `T` is a symbol? I would read the code as it being a type?

Reply:
Yes, naming could be improved. One thing is that I don't want this and `default_manifold_fallbacks` to be confused, so maybe `manifold_element_forwards` would be better.

`T` isn't necessarily a symbol; it was in the initial version of this macro but it turned out to be not expressive enough and I had to allow for more complicated `T`.

Reply:
I think `manifold_element_forwards` is distinguishable enough.

Ok I'll adapt the docs as well here, then.

Reply:
I somehow did not find tie last month, and now coming back I noticed that the `manifold_element_forwards function is documented with two arguments but actually has 3. The middle one is not documented.

Sure I could try to reverse engineer from its usage … but maybe I would be off with that – so what is it intended to be used for?

Reply:
Ah, maybe the supertype for `T`? Would it be useful to introduce a 2-parameter macro for the `_` case?